### PR TITLE
[BPF] test cleanup after switching back to iptables mode

### PR DESCRIPTION
**Pull request by tomastigera**
Also some fixes to remove all the maps.

## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
ebpf: fixed cleaning of programs and map when switching from ebpf to iptables mode.
```

## Reminder for the reviewer

Make sure that this PR has the correct labels and milestone set.

Every PR needs one `docs-*` label.

- `docs-pr-required`: This change requires a change to the documentation that has not been completed yet.
- `docs-completed`: This change has all necessary documentation completed.
- `docs-not-required`: This change has no user-facing impact and requires no docs.

Every PR needs one `release-note-*` label.

- `release-note-required`: This PR has user-facing changes. Most PRs should have this label.
- `release-note-not-required`: This PR has no user-facing changes.

Other optional labels:

- `cherry-pick-candidate`: This PR should be cherry-picked to an earlier release. For bug fixes only.
- `needs-operator-pr`: This PR is related to install and requires a corresponding change to the operator.

### CODE REVIEW

File: felix/bpf/cleanup.go [hunk 1]
```diff
@@ -28,7 +28,8 @@ func CleanUpCalicoPins(dir string) {
 		if err != nil {
 			return err
 		}
-		if strings.HasPrefix(info.Name(), "cali_") || strings.HasPrefix(info.Name(), "calico_") {
+		if strings.HasPrefix(info.Name(), "cali_") || strings.HasPrefix(info.Name(), "calico_") ||
```
Comment (fasaxc) [inline felix/bpf/cleanup.go:31]: Worth a utility function to share the match between these two files?

File: felix/fv/bpf_test.go [hunk 2]
```diff
@@ -941,6 +941,33 @@ func describeBPFTests(opts ...bpfTestOpt) bool {
...
 				})
+
+				It("should cleanup after we disable eBPF", func() {
+					By("Waiting for dp to get setup up")
+
+					ensureAllNodesBPFProgramsAttached(tc.Felixes)
+
+					By("Changing env and restarting felix")
+
+					tc.Felixes[0].SetEvn(map[string]string{"FELIX_BPFENABLED": "false"})
```
Comment (fasaxc) [inline felix/fv/bpf_test.go:952]:
```suggestion
					tc.Felixes[0].SetEnv(map[string]string{"FELIX_BPFENABLED": "false"})
```
typo in the utility func?

Comment (fasaxc) [reply felix/fv/bpf_test.go:952]: Didn't know we had that feature; FWIW, I added the `TestManagesBPF` flag to topology options for the "enable BPF" tests, you could use that (but this is fine too).

**Review by fasaxc — CHANGES_REQUESTED** [review]

File: felix/fv/bpf_test.go (hunk 2 above)
Comment (tomastigera) [reply felix/fv/bpf_test.go:952]: I am aware of that one, but I am happy to start in ebpf, but then i need to flip it to iptables.

File: felix/bpf/cleanup.go (hunk 1 above)
Comment (tomastigera) [reply felix/bpf/cleanup.go:31]: Ack, but the parsing is done differently in the test. and atm it is imo just fine.

**Comment by tomastigera** [comment]
@fasaxc the fv part of the tests passed

**Comment by fasaxc** [comment]
The test that's failing is an old BPF sidecar acceleration in iptables mode test, we must be cleaning up its map?

**Comment by tomastigera** [comment]
> The test that's failing is an old BPF sidecar acceleration in iptables mode test, we must be cleaning up its map?

yeah, that could be, there were some other collisions like that, let me check that.

**Review by fasaxc — APPROVED** [review]

**Pull request merged**